dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.ts

```diff
@@ -0,0 +1,76 @@
+
+const getLatestCompletedAmendmentCoverEndDate = (amendments: TfmFacilityAmendment[]): number | undefined => {
+  const completedAmendments = amendments.filter(({ status }) => status === 'Completed');
+  const latestAmendmentWithDefinedCoverEndDate = orderBy(completedAmendments, ['updatedAt'], ['desc']).find(({ coverEndDate }) => coverEndDate);
+  return latestAmendmentWithDefinedCoverEndDate?.coverEndDate ?? undefined;
+};
+
+const getLatestTfmFacilityValues = async (
+  facilityId: string,
+): Promise<{
+  coverEndDate: Date;
+  coverStartDate: Date;
+  dayCountBasis: number;
+  interestPercentage: number;
+}> => {
+  const tfmFacility = await TfmFacilitiesRepo.findOneByUkefFacilityId(facilityId);
+  if (!tfmFacility) {
+    throw new NotFoundError(`Failed to find a tfm facility with ukef facility id '${facilityId}'`);
+  }
+
+  const { coverEndDate: snapshotCoverEndDate, coverStartDate, dayCountBasis, interestPercentage } = tfmFacility.facilitySnapshot;
+  const latestAmendedCoverEndDate = tfmFacility.amendments ? getLatestCompletedAmendmentCoverEndDate(tfmFacility.amendments) : undefined;
+
+  const coverEndDate = latestAmendedCoverEndDate ?? snapshotCoverEndDate;
+  if (!coverEndDate) {
+    throw new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`);
+  }
+
+  if (!coverStartDate) {
+    throw new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`);
+  }
+
+  return {
+    coverEndDate: convertTimestampToDate(coverEndDate),
+    coverStartDate: convertTimestampToDate(coverStartDate),
+    dayCountBasis,
+    interestPercentage: new Big(interestPercentage).div(100).toNumber(),
+  };
+};
+
+const getNumberOfDaysRemainingInCoverPeriod = (reportPeriodStart: MonthAndYear, coverStartDate: Date, coverEndDate: Date): number => {
+  const currentReportPeriodStartMonthStart = startOfMonth(new Date(`${reportPeriodStart.year}-${reportPeriodStart.month}`));
+
+  if (isBefore(currentReportPeriodStartMonthStart, coverStartDate)) {
+    return differenceInDays(coverEndDate, coverStartDate);
+  }
+  return differenceInDays(coverEndDate, currentReportPeriodStartMonthStart);
+};
+
+/**
+ * Calculates the fixed fee for the current report period
+ * @param utilisation - The facility utilisation
+ * @param facilityId - The ukef facility id
+ * @param reportPeriod - The report period
+ * @returns The fixed fee for the current report period
+ */
+export const calculateFixedFee = async (utilisation: number, facilityId: string, reportPeriod: ReportPeriod): Promise<number> => {
+  const { coverEndDate, coverStartDate, dayCountBasis, interestPercentage } = await getLatestTfmFacilityValues(facilityId);
+
+  const numberOfDaysRemainingInCoverPeriod = getNumberOfDaysRemainingInCoverPeriod(reportPeriod.start, coverStartDate, coverEndDate);
+
+  return new Big(utilisation)
+    .mul(interestPercentage)
+    .mul(0.9) // use 90% of bank margin
```

Comment:
I think we should extract a named const for this and we should be explicit in our comment about the business reason we do this in case in the future someone looks at this and wonders why. Can probably just adapt from the ticket description:

> Bank fee adjustment - take 10% of the margin for their admin fee (fixed at 90%) multiplied by…